Quiz_maker

krassovki opened the pull request:
- added main
- added tests (all of them passed)
- test_get_given_amount_of_random_questions_with_wrong_keys_of_questions
  (line 54) trying to run toooo much time

krassovki committed:
Quiz_maker

- added main
- added tests (all of them passed)
- test_get_given_amount_of_random_questions_with_wrong_keys_of_questions
  (line 54) trying to run toooo much time

glock3 commented on Misha/Files/quiz_maker/quiz_maker_main.py:
```diff
@@ -0,0 +1,101 @@
+
+def get_answers_list(path):
+    if not os.path.exists(path):
+        raise FileNotFoundError('File not found')
```
Do you need this error?

krassovki replied:
Yes, to check if file exists

glock3 commented on Misha/Files/quiz_maker/quiz_maker_main.py:
```diff
@@ -0,0 +1,101 @@
+                        rand_index = random.randint(-1, number_of_questions_to_remove)
+                    questions.pop(rand_index)
+                except KeyError:
+                    raise KeyError
```
Do you need this raise?

krassovki replied:
Yes, to check if we have an error with dict keys

glock3 commented on Misha/Files/quiz_maker/quiz_maker_main.py:
```diff
@@ -0,0 +1,101 @@
+
+
+def get_needed_answers(key_answers, questions_keys):
+    if type(key_answers) != list or type(questions_keys) != list:
```
No need to check types here

krassovki replied:
okay(

glock3 commented on Misha/Files/quiz_maker/quiz_maker_test.py:
```diff
@@ -0,0 +1,115 @@
+        key_answers = 1488
+        questions_keys = [0, 1, 2, 3]
+        with self.assertRaises(TypeError) as context:
+            self.assertRaises(TypeError, get_needed_answers(key_answers, questions_keys))
```
You have two same asserts here

krassovki replied:
corrected

glock3 commented on Misha/Files/quiz_maker/quiz_maker_test.py:
```diff
@@ -0,0 +1,115 @@
+        key_answers = ['в', 'а', 'а', 'г']
+        questions_keys = 'string'
+        with self.assertRaises(TypeError) as context:
+            self.assertRaises(TypeError, get_needed_answers(key_answers, questions_keys))
```
You have two same asserts here

krassovki replied:
corrected

glock3 commented on Misha/Files/quiz_maker/quiz_maker_test.py:
```diff
@@ -0,0 +1,115 @@
+    def test_get_questions_with_wrong_paths(self):
+        path = 'wrong_path.txt'
+        with self.assertRaises(FileNotFoundError) as context:
+            self.assertRaises(FileNotFoundError, get_questions(path))
```
You have two same asserts here

krassovki replied:
corrected

glock3 commented on Misha/Files/quiz_maker/quiz_maker_test.py:
```diff
@@ -0,0 +1,115 @@
+                              2: 'Кто пидумал современную архитектуру ЭВМ?\nа. Джон фон-Нейман\nб. Карл Цузе\nв. '
+                                 'Бьярне Страусструп\nг. Илон Маск',
+                              3: 'Какое пиво самое крепкое?\nа. Жиуглевское\nб. Тетерев\nв. Московское\nг. Опилля'}
+        questions = get_questions(path)
```
add tests for this method please

krassovki replied:
done in this version

glock3 commented on Misha/Files/quiz_maker/quiz_maker_test.py:
```diff
@@ -0,0 +1,115 @@
+    def test_get_answer_list_with_right_paths(self):
+        path = 'answers.txt'
+        expected_answers = ['в', 'а', 'а', 'г']
+        answers = get_answers_list(path)
```
Add tests for this method please

krassovki replied:
done in this version

glock3 commented on Misha/Files/quiz_maker/quiz_maker_test.py:
```diff
@@ -0,0 +1,115 @@
+                     2: 'Кто пидумал современную архитектуру ЭВМ?\nа. Джон фон-Нейман\nб. Карл Цузе\nв. '
+                        'Бьярне Страусструп\nг. Илон Маск',
+                     3: 'Какое пиво самое крепкое?\nа. Жиуглевское\nб. Тетерев\nв. Московское\nг. Опилля'}
+        questions = get_given_amount_of_random_questions(number_of_questions, questions)
```
Add tests for this function

krassovki replied:
done in this version

glock3 commented on Misha/Files/quiz_maker/quiz_maker_main.py:
```diff
@@ -0,0 +1,101 @@
+        text = ''
+        for line in f:
+            text += line
+        f.close()
```
Please close file in final block or use with-resource statement

krassovki replied:
why cant i close it after finishin using it?

krassovki committed:
quiz_maker

- deleted useless tests
- corrected some tests with double assert
- deleted check-ins for type of given arg

krassovki committed:
quiz_maker

- Corrected test that did not want to start
- Deleted useless raises

krassovki committed:
quiz_maker

- Corrected all asserts
